SimTKcommon/Simulation/include/SimTKcommon/internal/StateImpl.h

```diff
@@ -636,7 +636,7 @@ class SimTK_SimTKCOMMON_EXPORT StateImpl {
 
     StateImpl& operator=(const StateImpl& src);
 
-    ~StateImpl() {}   // default destructor
+    ~StateImpl() {delete stateLock;}
 
     // Copies all the variables but not the cache.
     StateImpl* clone() const {return new StateImpl(*this);}
```

Comment:
I wasn't sure should happen if those things happen - I also made a note of it in the copy construction in State.cpp:316